X/xtensor_io/build_tarballs.jl

```diff
@@ -0,0 +1,57 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, BinaryBuilderBase, Pkg
+
+name = "xtensor_io"
+version = v"0.13.0"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/xtensor-stack/xtensor-io", "ffada938383b0f24c9e0b07cea7d5780057e1d96"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd ${WORKSPACE}/srcdir/xtensor-io
+
+# There is a known problem with HighFive, apparently a HighFive issue.
+# HighFive is close to releasing version 3.0.0, we will include it then.
+# An alternative would be to use an older version of HighFive.
+# Apparently 2.1.1 works, 2.2.2 doesn't.
+# -DHAVE_HighFive=ON
+cmake -B build -G Ninja \
+    -DCMAKE_INSTALL_PREFIX=${prefix} \
+    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DBUILD_SHARED_LIBS=ON \
+    -DDOWNLOAD_GBENCHMARK=OFF \
+    -DHAVE_Blosc=ON \
+    -DHAVE_OIIO=ON \
+    -DHAVE_ZLIB=ON
+cmake --build build --parallel ${nproc}
+cmake --install build
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [AnyPlatform()]
+
+# The products that we will ensure are always built
+products = Product[
+    # This is a header-only library without any binary products
+    FileProduct("include/xtensor-io/xtensor-io.hpp", :xtensor_io),
```

Comment:
What does this mean to be a header-only library but have a CMake flag of `-DBUILD_SHARED_LIBS=ON`?

Reply:
I don't know. The package does store some CMake flags and compiler flags in CMake config files and downstream packages will look at them... Maybe it helps ensure that `-fPIC` is used etc.?